### ISSUE TITLE
Input without label does not produce error

### ISSUE THREAD

**Issue by pennmeg** [dequelabs/axe-core#4384]
### Product

axe Linter

### Product Version

4.8.3

### Latest Version

- [X] I have tested the issue with the latest version of the product

#### Expectation
That the VS Code extension or GitHub Action linter would flag the following code as an error based on there is no explicit relationship between the label and the input. 

```
     <label>First Name</label>
     <input type="text" />
```

#### Actual
The code does not produce an error in either VS Code or the linter, despite there being no ID/for, ID/aria-labelledby, or other marker to make a relationship between the input and label. Believe this should be an error based upon the [Label rule](https://dequeuniversity.com/rules/axe/4.8/label)

```
     <label>First Name</label>
     <input type="text" />
```

#### How to Reproduce
Created a React application, and have the following code in the App.js file.

```
  <label>First Name</label>
   <input type="text" />
```

[Code snippet in context of App.js file.](https://github.com/pennmeg/a11y-actions/blob/main/src/App.js#L21)

#### Additional context
I've also tried removing the label element entirely to see if a single input could trigger an error and was not successful. 

Appreciate feedback if I'm interpreting the [label rule](https://dequeuniversity.com/rules/axe/4.8/label) correctly or this example is expected behavior

**Comment by pennmeg** [comment]
> Thanks for the issue. Since axe-linter runs on non-complete trees of HTML (things like React, Vue, and Angular components, as well as just random HTML code without a full page) it can't know if the code in question will be used inside a component that would then wrap the input with a label in the final output. Because of this many relational based rules won't run unless it detects you're running in a full page. That is, there's an `<html>` or `<body>` element at the root. If you wrap the code in question in an either of those you should see an error for the missing label.

Appreciate the added context and suggestion. When I created a basic `index.html` page with the same code provided here I did in fact see the error `Ensures every form element has a label`. Will close out issue.
